File: docs/sharepoint/codesnippet/VisualBasic/upgradedeploymentstep/sharepointcommands/commands.vb
'<Snippet4>
Imports System
Imports System.IO
Imports Microsoft.SharePoint
Imports Microsoft.SharePoint.Administration
Imports Microsoft.VisualStudio.SharePoint.Commands

Namespace Contoso.DeploymentSteps.Upgrade

    Friend Class Commands

        ' Determines whether the specified solution has been deployed to the local SharePoint server.
        <SharePointCommand("Contoso.Commands.IsSolutionDeployed")> _
        Private Function IsSolutionDeployed(ByVal context As ISharePointCommandContext, ByVal solutionName As String) As Boolean
            Dim solution As SPSolution = SPFarm.Local.Solutions(solutionName)
            Return solution IsNot Nothing
        End Function

        ' Upgrades the specified solution to the local SharePoint server.
        '<Snippet5>
        <SharePointCommand("Contoso.Commands.UpgradeSolution")> _
        Private Sub UpgradeSolution(ByVal context As ISharePointCommandContext, ByVal fullWspPath As String)
            Dim solution As SPSolution = SPFarm.Local.Solutions(Path.GetFileName(fullWspPath))
            If solution Is Nothing Then
                Throw New InvalidOperationException("The solution has not been deployed.")
            End If
            solution.Upgrade(fullWspPath)
        End Sub
        '</Snippet5>

    End Class
End Namespace
'</Snippet4>

File: docs/sharepoint/codesnippet/CSharp/UpgradeDeploymentStep/SharePointCommands/Commands.cs
//<Snippet4>
using System;
using System.IO;
using Microsoft.SharePoint;
using Microsoft.SharePoint.Administration;
using Microsoft.VisualStudio.SharePoint.Commands;

namespace Contoso.DeploymentSteps.Upgrade
{
    internal class Commands
    {
        // Determines whether the specified solution has been deployed to the local SharePoint server.
        [SharePointCommand("Contoso.Commands.IsSolutionDeployed")]
        private bool IsSolutionDeployed(ISharePointCommandContext context, string solutionName)
        {
            SPSolution solution = SPFarm.Local.Solutions[solutionName];
            return solution != null;
        }

        // Upgrades the specified solution to the local SharePoint server.
        //<Snippet5>
        [SharePointCommand("Contoso.Commands.UpgradeSolution")]
        private void UpgradeSolution(ISharePointCommandContext context, string fullWspPath)
        {
            SPSolution solution = SPFarm.Local.Solutions[Path.GetFileName(fullWspPath)];

            if (solution == null)
            {
                throw new InvalidOperationException("The solution has not been deployed.");
            }

            solution.Upgrade(fullWspPath);
        }
        //</Snippet5>
    }
}
//</Snippet4>
